specification/automation/resource-manager/Microsoft.Automation/stable/2018-01-15/dscNode.json

```diff
@@ -0,0 +1,957 @@
+            "type": "string",
+            "description": "The report id."
+          },
+          {
+            "$ref": "./definitions.json#/parameters/SubscriptionIdParameter"
+          },
+          {
+            "$ref": "./definitions.json#/parameters/ApiVersionParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "OK",
+            "schema": {
+              "type": "file"
+            }
+          },
+          "default": {
+            "description": "Automation error response describing why the operation failed.",
+            "schema": {
+              "$ref": "./definitions.json#/definitions/ErrorResponse"
+            }
+          }
+        }
+      }
+    }
+  },
+  "definitions":{
+    "DscNodeReportListResult": {
+      "properties": {
+        "value": {
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/DscNodeReport"
+          },
+          "description": "Gets or sets a list of dsc node reports."
+        },
+        "nextLink": {
+          "type": "string",
+          "description": "Gets or sets the next link."
+        }
+      },
+      "description": "The response model for the list dsc nodes operation."
+    },
+    "DscReportResourceNavigation": {
+      "properties": {
+        "resourceId": {
+          "type": "string",
+          "description": "Gets or sets the ID of the resource to navigate to."
+        }
+      },
+      "description": "Navigation for DSC Report Resource."
+    },
+    "DscMetaConfiguration": {
+      "properties": {
+        "configurationModeFrequencyMins": {
+          "type": "integer",
+          "format": "int32",
+          "description": "Gets or sets the ConfigurationModeFrequencyMins value of the meta configuration."
+        },
+        "rebootNodeIfNeeded": {
+          "type": "boolean",
+          "description": "Gets or sets the RebootNodeIfNeeded value of the meta configuration."
+        },
+        "configurationMode": {
+          "type": "string",
+          "description": "Gets or sets the ConfigurationMode value of the meta configuration."
+        },
+        "actionAfterReboot": {
+          "type": "string",
+          "description": "Gets or sets the ActionAfterReboot value of the meta configuration."
+        },
+        "certificateId": {
+          "type": "string",
+          "description": "Gets or sets the CertificateId value of the meta configuration."
+        },
+        "refreshFrequencyMins": {
+          "type": "integer",
+          "format": "int32",
+          "description": "Gets or sets the RefreshFrequencyMins value of the meta configuration."
+        },
+        "allowModuleOverwrite": {
+          "type": "boolean",
+          "description": "Gets or sets the AllowModuleOverwrite value of the meta configuration."
+        }
+      },
+      "description": "Definition of the DSC Meta Configuration."
+    },
+    "DscNodeReport": {
```

Comment:
>DscNodeReport [](start = 5, length = 13)

Some properties in this definition, I believe could be represented with enums

Reply:
Yup. Agreed. Let me check the implementation to verify.

Reply:
Same as above: We aren't planning to enum this currently as the things being returned here are not finalized yet and are subject to change.